src/Module/AddBookingToCartHandler.php

```diff
@@ -151,15 +186,58 @@ protected function _addToCart(BookingInterface $booking)
         // Get the cart item data keys from the cart item config
         $eddBkKey     = $this->_containerGetPath($this->cartItemConfig, ['data', 'eddbk_key']);
         $bookingIdKey = $this->_containerGetPath($this->cartItemConfig, ['data', 'booking_id_key']);
+        $priceIdKey   = $this->_containerGetPath($this->cartItemConfig, ['data', 'price_id_key']);
+
+        // Find the price ID
+        $service  = $this->_getServiceById($serviceId);
+        $lengths  = $this->_containerGet($service, 'session_lengths');
+        $duration = $booking->getDuration();
+        $priceId  = null;
+        foreach ($lengths as $_idx => $_lengthInfo) {
+            $_length = (int) $this->_containerGet($_lengthInfo, 'sessionLength');
+
+            if ($_length === $duration) {
+                $priceId = $_idx;
+                break;
+            }
+        }
 
         // Create the cart item data
         $data = [
-            $eddBkKey => [
+            $priceIdKey => $priceId,
+            $eddBkKey   => [
                 $bookingIdKey => $bookingId,
             ],
         ];
 
         // Add the service (EDD Download) to the cart with the additional EDD Bookings data
         $this->eddCart->add($serviceId, $data);
     }
+
+    /**
+     * Retrieves a booking by ID.
+     *
+     * @since [*next-version*]
+     *
+     * @param int|string|Stringable $serviceId The ID of the service to retrieve.
+     *
+     * @return array|stdClass|ArrayAccess|ContainerInterface|null The service instance, or null if no service was
+     *                                                            found for the given ID.
+     */
+    protected function _getServiceById($serviceId)
+    {
+        $b = $this->exprBuilder;
+
+        $services = $this->servicesSelectRm->select(
+            $b->and(
+                $b->eq($b->ef('service', 'id'), $b->lit($serviceId))
+            )
+        );
+
+        if ($this->_countIterable($services) !== 1) {
```

Comment:
I've noticed this in several other places. If for some reason the returned set is large, and it is not countable, then that whole set will be iterated over, just to fall back.

Reply:
True. Any suggestions for change?

Reply:
What happens if you just use `reset()`, without counting? If there's at least one item, it will be used. If there isn't any, `false` is returned. So, maybe something like this?

```php
return reset($services) ?: null;
```

What do you think?

Reply:
Cool. I thought the shorthand `?:` was a PHP7+ thing, but I searched the manual and turns out its 5.3+ 😃

Reply:
We're just stuck thinking about WP all the time.